src/main/java/com/example/weatherwaveapi/controller/WeatherController.java

```diff
@@ -12,45 +13,73 @@
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 
 @RequiredArgsConstructor
 @RestController
 @RequestMapping("/weather")
 @Slf4j
 public class WeatherController {
+    private static final Integer ZERO = 0;
+    private static final Integer ONE = 1;
+    private static final Integer TWO = 2;
     private static final String REQUEST_MESSAGE = "Request contain serviceApiEnumValue: {}";
     private static final String INVALID_INPUT_MESSAGE = "Invalid input for parameter %s. Please provide a valid value.";
 
-    private final OpenWeatherService openWeatherService;
+    private final WeatherRouterService weatherRouterService;
 
-    @GetMapping(value = "/{city}", produces = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<WeatherResponse> getWeatherInSelectedCity(@PathVariable(name = "city") String city,
-                                                                    @RequestParam(name = "serviceApi", defaultValue = "OPEN_WEATHER_MAP", required = false) ServiceApiEnum inputOfServiceApi) {
-        return getWeatherResponse(List.of(city), inputOfServiceApi);
-    }
-
-    @GetMapping(value = "/cities", produces = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<WeatherResponse> getWeatherInSelectedCities(
-            @RequestParam(value = "cities") List<String> cities,
-            @RequestParam(value = "serviceApi", defaultValue = "OPEN_WEATHER_MAP", required = false) ServiceApiEnum inputOfServiceApi
+    @GetMapping(value = "/params", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<WeatherResponse> getWeatherData(@RequestParam(value = "cities", required = false) List<String> cities,
+                                                          @RequestParam(value = "zipCodes", required = false) List<String> zipCodes,
+                                                          @RequestParam(value = "coordinates", required = false) List<String> coordinates,
+                                                          @RequestParam(value = "serviceApi", defaultValue = "OPEN_WEATHER_MAP", required = false) ServiceApiEnum inputOfServiceApi
     ) {
-        return getWeatherResponse(cities, inputOfServiceApi);
+
+        List<CoordinateWeatherRequest> coordinateRequests = createCoordinateRequests(coordinates);
+        List<ZipCodeWeatherRequest> zipCodeWeatherRequests = createZipCodeRequest(zipCodes);
+
+        WeatherRequest weatherRequest = WeatherRequest.builder().cities(cities).zipcode(zipCodeWeatherRequests).coordinates(coordinateRequests).service(inputOfServiceApi).build();
+        log.info(REQUEST_MESSAGE, weatherRequest);
+        WeatherService weatherService = weatherRouterService.getWeatherBySelectedService(inputOfServiceApi);
+        WeatherResponse weatherResponse = weatherService.getWeather(weatherRequest);
```

Comment:
```suggestion
        WeatherResponse weatherResponse = weatherRouterService.getWeatherBySelectedService(inputOfServiceApi).getWeather(weatherRequest);
```
Having a record in 2 lines does not give us any advantages

Reply:
I did it